custom_components/opensprinkler/__init__.py

```diff
@@ -38,8 +49,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     password = entry.data.get(CONF_PASSWORD)
     try:
         controller = OpenSprinkler(url, password)
-        await hass.async_add_executor_job(controller.refresh)
-        coordinator = OpenSprinklerCoordinator(hass, controller)
+        controller.refresh_on_update = False
+
+        async def async_update_data():
+            """Fetch data from OpenSprinkler."""
+            print("refreshing OpenSprinkler data")
+            async with async_timeout.timeout(TIMEOUT):
+                await hass.async_add_executor_job(controller.refresh)
+                if not controller._state:
+                    raise UpdateFailed("Error fetching OpenSprinkler state")
+
+                return controller._state
+
+        scan_interval = entry.data.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
```

Comment:
I guess this just get the default ATM? Just FYI, this conf will need to go into `OptionFlow` rather than `ConfigFlow` to go into core. So a refactor will be needed later

Reply:
Ah OK. So what changes do I need to make here?

Reply:
To be honest I am not sure as I have never done `OptionFlow` before but this was a comment on the core PR as I initially had a run seconds config and I assume this will apply for the scan interval as well. 

But I don't think it is a small change so its probably worth to get this in first.

Reply:
I think it may be `entry.options` is all versus `entry.data`. Want me to tweak that?

Reply:
I don't think its that simple as `entry.options` will be empty here, unless options `OptionFlow` goes through the whole setup flow again. Since it is an unknown let's keep it as it is and do it properly when you do the `OptionFlow`

Reply:
Well, the values in `entry.options` come from the options flow yes. I'm not skilled enough to do that but at least it's looking in the right place now.